R/HS_PGA_num.R

```diff
@@ -0,0 +1,39 @@
+#' Calculates HS-PGA scores based on abscess, draining fistula, and inflammatory nodule counts.
+#'
+#' @param abscess_fistula Integer vector representing the sum of abscess and draining fistula counts.
+#' @param inflammatory_nodule Integer vector representing the inflammatory nodule count.
+#' @param non_inflammatory_nodule Integer vector representing the non-inflammatory nodule count.
+#' @return An integer vector representing the HS-PGA scores (1: Clean, 2: Minimal, 3: Mild, 4: Moderate, 5: Severe, 6: Very Severe).
+#' @export
+#' @examples
+#' 
+#' abscess_fistula <- c(0, 0, 1, 0, 1, 2, 6)
+#' inflammatory_nodule <- c(0, 0, 0, 3, 5, 8, 12)
+#' non_inflammatory_nodule <- c(0, 1, 0, 0, 0, 0, 0)
+#'
+#' hs_pga_num(abscess_fistula, inflammatory_nodule, non_inflammatory_nodule)
+#'
+
+hs_pga_num <- function(abscess_fistula, inflammatory_nodule, non_inflammatory_nodule) {
+  
+  # Assertions using checkmate package
+  checkmate::assert_integerish(abscess_fistula, lower = 0, any.missing = FALSE, null.ok = FALSE)
+  checkmate::assert_integerish(inflammatory_nodule, lower = 0, len = length(abscess_fistula), any.missing = FALSE, null.ok = FALSE)
+  checkmate::assert_integerish(non_inflammatory_nodule, lower = 0, len = length(abscess_fistula), any.missing = FALSE, null.ok = FALSE)
+  
+  # Initialize the HS-PGA scores vector for use in conditionals
+  hs_pga_scores <- integer(length(abscess_fistula))
```

Comment:
This defaults to zeros which may cause issues in case of out of range values, etc.  Can you please replace this with:

```
hs_pga_scores <- rep(NA_integer_, length(abscess_fistula))
```

That way, the default output becomes `NA` instead of `0`.

Reply:
Replaced with #9